[Outlook] (preview) Add-in support in the new Outlook on Windows (#3837)
OfficeDev/office-js-docs-pr: docs/tutorials/migrate-vsto-to-office-add-in-shared-code-library-tutorial.md

---
title: 'Tutorial: Share code between both a VSTO Add-in and an Office Add-in by using a shared code library'
description: Tutorial on how to share code between a VSTO Add-in and an Office Add-in.

> ms.date: 02/02/2023

Reviewer: ```suggestion
ms.date: 02/07/2023
```

ms.prod: non-product-specific
ms.localizationpriority: high
---

# Tutorial: Share code between both a VSTO Add-in and an Office Add-in with a shared code library

> Visual Studio Tools for Office (VSTO) Add-ins are great for extending Office to provide solutions for your business or others. They've been around for a long time and there are thousands of solutions built with VSTO. However, they only run on Office on Windows. You can't run VSTO Add-ins on Mac, online, or mobile platforms.

Reviewer: ```suggestion
Visual Studio Tools for Office (VSTO) Add-ins are great for extending Office to provide solutions for your business or others. They've been around for a long time and there are thousands of solutions built with VSTO. However, they only run on Office on Windows. You can't run VSTO Add-ins on Mac, on the web, or on mobile platforms.
```

1. In the previous code, enter the **sslPort** number you saved previously from the **launchSettings.json** file.

In the previous code, the returned string will be processed to replace carriage return line feeds with `<br>` HTML tags. You may occasionally run into situations where a return value that works perfectly fine for .NET in the VSTO Add-in will need to be adjusted on the Office Add-in side to work as expected. In this case, the REST API and shared class library are only concerned with returning the string. The `showUnicode()` function is responsible for formatting return values correctly for presentation.

### Allow CORS from the Office Add-in

The Office.js library requires CORS on outgoing calls, such as the one made from the `ajax` call to the REST API server. Use the following steps to allow calls from the Office Add-in to the REST API.

1. In **Solution Explorer**, select the **CellAnalyzerOfficeAddinWeb** project.

> 1. From the **View** menu, choose **Properties Window** (if the window isn't already displayed).

Reviewer 2: ```suggestion
1. From the **View** menu, choose **Properties Window**, if the window isn't already displayed.
```
